File: java/katas/src/main/java/PrimeNumber.java
/**
 * PROBLEM
 * Given a number i return if it is prime.
 *
 * @author Mike Lowe
 */
public class PrimeNumber {

    // don't want instances
    private PrimeNumber() {
    }

    public static boolean isPrime(int i) {
        if (i <= 1) {
            return false;
        }
        int mid = i / 2;
        for (int j = 2; j <= mid; j++) {
            if (i % j == 0) {
                return false;
            }
        }
        return true;
    }

}

File: javascript/katas/mwhelan/src/PrimeNumber.js
/*
 * PROBLEM
 * Given a number n return if it is prime.
 */

'use strict';

const PrimeNumber = {

  isPrime: (n) => {
    if (n < 2) return false;
    for (let i = 2, mid = n / 2; i <= mid; i++) {
      if (n % i === 0) {
        return false;
      }
    }
    return true;
  }

};

module.exports = PrimeNumber;